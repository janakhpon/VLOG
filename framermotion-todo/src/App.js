import { useState } from "react";
import { motion, Reorder, AnimatePresence, LayoutGroup } from "framer-motion";

function App() {
  const [tasks, setTasks] = useState([
    {
      id: 1,
      description:
        "Your mouth is a revolver, firing bullets in the sky (Mh-mh-mh-mh)",
      date: "Sat Jan 14 2022",
    },
    {
      id: 2,
      description:
        "People like us, we dont Need that much, just some One that starts, starts the spark in our bonfire hearts",
      date: "Sat Jan 13 2022",
    },
    {
      id: 3,
      description:
        "This world is getting colder, strangers passing by (Mh-mh-mh-mh) No one offers you a shoulder, no one looks you in the eye, eye, eye (Mh-mh-mh-mh)",
      date: "Sat Jan 15 2022",
    },
    {
      id: 4,
      description:
        "Days like these lead to Nights like this lead to Love like ours You light the spark in my bonfire heart",
      date: "Sat Jan 12 2022",
    },
  ]);
  const [newTask, setNewTask] = useState({
    description: "",
    date: "",
  });
  const [lastkey, setLastkey] = useState(5);

  const handleChange = (e) => {
    e.preventDefault();
    const day = new Date();
    const { value } = e.target;
    setNewTask({
      date: day.toDateString(),
      description: value,
    });
  };

  const removeTask = (id) => {
    setTasks([...tasks.filter((t) => t.id !== id)]);
  };

  const handleEnterPress = (e) => {
    if (e.keyCode === 13 && e.shiftKey === false) {
      e.preventDefault();
      setTasks([{ ...newTask, id: lastkey + 1 }, ...tasks]);
      setLastkey(lastkey + 1);
      setNewTask({
        description: "",
        date: "",
      });
    }
  };

  return (
    <div className="h-screen bg-slate-300">
      <div className="px-10 py-10 md:container md:mx-auto">
        {/* Simple ul list */}
        {/* <LayoutGroup>
          <div className="max-w-3xl mx-auto my-6">
            <motion.form
              initial={{
                y: "-100",
                opacity: 0,
              }}
              animate={{
                y: 0,
                opacity: 1,
                transition: {
                  duration: 1,
                  delay: 0.2,
                  type: "spring",
                  bounce: 0.85,
                },
              }}
              whileTap={{
                scale: 0.95,
              }}
              whileHover={{
                y: -5,
                transition: {
                  type: "spring",
                  bounce: 0.85,
                },
              }}
              className="w-full"
            >
              <textarea
                className="
                    form-control
                    block
                    w-full
                    px-2
                    py-1.5
                    text-base
                    font-normal
                    text-gray-700
                    bg-white bg-clip-padding
                    border border-solid border-gray-300
                    rounded
                    m-0
                    focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
                id="exampleFormControlTextarea1"
                rows="4"
                placeholder="Your new task"
                value={newTask.description}
                onChange={handleChange}
                onKeyDown={handleEnterPress}
              ></textarea>
            </motion.form>
          </div>
          <motion.ul
            className="overflow-y-auto"
            style={{ maxHeight: "80vh" }}
          >
            <AnimatePresence>
              {tasks.map((task, i) => {
                return (
                  <motion.li
                    initial={{ x: -100, opacity: 0 }}
                    animate={{
                      x: 0,
                      opacity: 0.8,
                      transition: {
                        type: "spring",
                        stiffness: 150,
                        duration: 1,
                        delay: i / 4,
                      },
                    }}
                    exit={{
                      x: 50,
                      opacity: 0,
                      transition: {
                        ease: "easeInOut",
                        duration: 0.25,
                        delay: 0.15,
                      },
                    }}
                    whileHover={{
                      opacity: 1,
                      scale: 1.03,
                      transition: {
                        type: "spring",
                        bounce: 0.85,
                      },
                    }}
                    key={task.id}
                    className="max-w-3xl p-4 mx-auto my-4 rounded-lg shadow-lg bg-slate-200"
                  >
                    <h4 className="text-lg font-semibold text-gray-800">
                      {task.description}
                    </h4>
                    <p className="mt-2 text-sm font-semibold text-gray-600">
                      Created on {task.date}
                    </p>
                    <div className="flex justify-end space-x-2">
                      <svg
                        xmlns="http://www.w3.org/2000/svg"
                        className="w-6 h-6"
                        fill="none"
                        viewBox="0 0 24 24"
                        stroke="currentColor"
                      >
                        <path
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth={2}
                          d="M5 13l4 4L19 7"
                        />
                      </svg>
                      <svg
                        xmlns="http://www.w3.org/2000/svg"
                        className="w-6 h-6"
                        fill="none"
                        viewBox="0 0 24 24"
                        stroke="currentColor"
                      >
                        <path
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth={2}
                          d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
                        />
                      </svg>
                      <svg
                        xmlns="http://www.w3.org/2000/svg"
                        className="w-6 h-6 cursor-pointer"
                        fill="none"
                        viewBox="0 0 24 24"
                        stroke="currentColor"
                        onClick={() => removeTask(task.id)}
                      >
                        <path
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth={2}
                          d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                        />
                      </svg>
                    </div>
                  </motion.li>
                );
              })}
            </AnimatePresence>
          </motion.ul>
        </LayoutGroup> */}

        {/* Reorder component */}

        <LayoutGroup>
          <div className="max-w-3xl mx-auto my-6">
            <motion.form
              initial={{
                y: "-100",
                opacity: 0,
              }}
              animate={{
                y: 0,
                opacity: 1,
                transition: {
                  duration: 1,
                  delay: 0.2,
                  type: "spring",
                  bounce: 0.85,
                },
              }}
              whileTap={{
                scale: 0.95,
              }}
              whileHover={{
                y: -5,
                transition: {
                  type: "spring",
                  bounce: 0.85,
                },
              }}
              className="w-full"
            >
              <textarea
                className="
                    form-control
                    block
                    w-full
                    px-2
                    py-1.5
                    text-base
                    font-normal
                    text-gray-700
                    bg-white bg-clip-padding
                    border border-solid border-gray-300
                    rounded
                    m-0
                    focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
                id="exampleFormControlTextarea1"
                rows="4"
                placeholder="Your new task"
                value={newTask.description}
                onChange={handleChange}
                onKeyDown={handleEnterPress}
              ></textarea>
            </motion.form>
          </div>
          <Reorder.Group
            axis="y"
            className="overflow-y-auto"
            style={{ maxHeight: "80vh" }}
            values={tasks}
            onReorder={setTasks}
          >
            <AnimatePresence>
              {tasks.map((task, i) => {
                return (
                  <Reorder.Item
                    initial={{ x: -100, opacity: 0 }}
                    animate={{
                      x: 0,
                      opacity: 0.8,
                      transition: {
                        type: "spring",
                        stiffness: 150,
                        duration: 1,
                        delay: i / 4,
                      },
                    }}
                    exit={{
                      x: 50,
                      opacity: 0,
                      transition: {
                        ease: "easeInOut",
                        duration: 0.25,
                        delay: 0.15,
                      },
                    }}
                    whileHover={{
                      opacity: 1,
                      scale: 1.03,
                      transition: {
                        type: "spring",
                        bounce: 0.85,
                      },
                    }}
                    key={task.id}
                    value={task}
                    className="max-w-3xl p-4 mx-auto my-4 rounded-lg shadow-lg bg-slate-200"
                  >
                    <h4 className="text-lg font-semibold text-gray-800">
                      {task.description}
                    </h4>
                    <p className="mt-2 text-sm font-semibold text-gray-600">
                      Created on {task.date}
                    </p>
                    <div className="flex justify-end space-x-2">
                      <svg
                        xmlns="http://www.w3.org/2000/svg"
                        className="w-6 h-6"
                        fill="none"
                        viewBox="0 0 24 24"
                        stroke="currentColor"
                      >
                        <path
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth={2}
                          d="M5 13l4 4L19 7"
                        />
                      </svg>
                      <svg
                        xmlns="http://www.w3.org/2000/svg"
                        className="w-6 h-6"
                        fill="none"
                        viewBox="0 0 24 24"
                        stroke="currentColor"
                      >
                        <path
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth={2}
                          d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
                        />
                      </svg>
                      <svg
                        xmlns="http://www.w3.org/2000/svg"
                        className="w-6 h-6 cursor-pointer"
                        fill="none"
                        viewBox="0 0 24 24"
                        stroke="currentColor"
                        onClick={() => removeTask(task.id)}
                      >
                        <path
                          strokeLinecap="round"
                          strokeLinejoin="round"
                          strokeWidth={2}
                          d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                        />
                      </svg>
                    </div>
                  </Reorder.Item>
                );
              })}
            </AnimatePresence>
          </Reorder.Group>
        </LayoutGroup>
      </div>
    </div>
  );
}

export default App;
