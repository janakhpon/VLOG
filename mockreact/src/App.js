import { useEffect, useState } from "react";
import axios from "axios";

function App() {
  const [tasks, setTasks] = useState([]);
  const [newTask, setNewTask] = useState({
    title: "",
    text: "",
  });
  useEffect(() => {
    let isApiSubscribed = true;
    axios.get("http://localhost:3001/api/v1/todos").then((res) => {
      if (isApiSubscribed) setTasks(res.data);
    });
    return () => {
      isApiSubscribed = false;
    };
  }, []);

  const handleChange = (e) => {
    e.preventDefault();
    const day = new Date();
    const { value } = e.target;
    setNewTask({
      title: day.toDateString(),
      text: value,
    });
  };

  const handleEnterPress = async (e) => {
    if (e.keyCode === 13 && e.shiftKey === false) {
      e.preventDefault();
      const res = await axios.post(
        "http://localhost:3001/api/v1/todos",
        newTask
      );
      setTasks([res.data, ...tasks]);
      setNewTask({
        title: "",
        text: "",
      });
    }
  };

  const handleComplete = async (task) => {
    const resp = await axios.put(
      `http://localhost:3001/api/v1/todos/${task.id}`,
      { ...task, completed: true }
    );
    setTasks([...tasks.map((t) => (t.id === task.id ? resp.data : t))]);
  };

  const handleDelete = async (id) => {
    await axios.delete(`http://localhost:3001/api/v1/todos/${id}`);
    setTasks([...tasks.filter((t) => t.id !== id)]);
  };

  return (
    <div className="h-screen bg-slate-300">
      <div className="px-2 py-2 md:px-10 md:py-10 md:container md:mx-auto">
        <div className="max-w-3xl mx-auto my-0 md:my-6">
          <form className="w-full">
            <textarea
              className="
                    form-control
                    block
                    w-full
                    px-2
                    py-1.5
                    text-base
                    font-normal
                    text-gray-700
                    bg-white bg-clip-padding
                    border border-solid border-gray-300
                    rounded
                    m-0
                    focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
              id="exampleFormControlTextarea1"
              rows="4"
              placeholder="Your new task"
              value={newTask.text}
              onChange={handleChange}
              onKeyDown={handleEnterPress}
            ></textarea>
          </form>
        </div>
        <ul
          className="overflow-y-auto"
          style={{ maxHeight: "80vh" }}
          data-testid="taskList"
        >
          {tasks?.map((task, i) => {
            return (
              <li
                key={i}
                className="max-w-3xl p-4 mx-auto my-4 rounded-lg shadow-lg bg-slate-200"
                data-testid={`listItem${i}`}
              >
                <h4 className="text-lg font-semibold text-gray-800">
                  {task.text}
                </h4>
                <p
                  className="mt-2 text-sm font-semibold text-gray-600"
                  label={task.completed ? "Completed" : "Not Completed"}
                  data-testid={`listItemCompleted${i}`}
                >
                  {task.completed ? "Completed" : "Not Completed"}
                </p>
                <div className="flex justify-end space-x-2">
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    className="w-6 h-6"
                    fill="none"
                    viewBox="0 0 24 24"
                    stroke="currentColor"
                    data-testid={`completeButton${i}`}
                    onClick={() => handleComplete(task)}
                  >
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth={2}
                      d="M5 13l4 4L19 7"
                    />
                  </svg>
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    className="w-6 h-6"
                    fill="none"
                    viewBox="0 0 24 24"
                    stroke="currentColor"
                  >
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth={2}
                      d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
                    />
                  </svg>
                  <svg
                    xmlns="http://www.w3.org/2000/svg"
                    className="w-6 h-6 cursor-pointer"
                    fill="none"
                    viewBox="0 0 24 24"
                    stroke="currentColor"
                    data-testid={`deleteButton${i}`}
                    onClick={() => handleDelete(task.id)}
                  >
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth={2}
                      d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                    />
                  </svg>
                </div>
              </li>
            );
          })}
        </ul>
      </div>
    </div>
  );
}

export default App;
