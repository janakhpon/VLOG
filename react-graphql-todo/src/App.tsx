import { ChangeEvent, FormEvent, useState, useEffect } from 'react'
import { useQuery, useMutation } from '@apollo/react-hooks'
import Pagination from './Components/Pagination'
import DisplayTasks from './Components/Tasks'
import InteractiveInputComponent from './Components/InputComponent'
import GQL_GETCOUNT from './graphql/getCount'
import GQL_GETLIST from './graphql/getList'
import GQL_GETSEARCHLIST from './graphql/getSearchedList'
import GQL_CREATETASK from './graphql/createTask'
import GQL_UPDATETASK from './graphql/updateTask'
import GQL_DELETETASK from './graphql/deleteTask'
import { TaskType, CreateTask, SortType } from './Types/task.type'
import { calculatePagination, days } from './utils/task.common'
const App = () => {
  const [tasks, setTasks] = useState<TaskType[]>([])
  const [searchedTasks, setSearchedTasks] = useState<TaskType[]>([])
  const [task, setTask] = useState<CreateTask>({
    title: "",
    text: ""
  })
  const [search, setSearch] = useState<string>("")
  const [page, setPage] = useState<number>(1)
  const [count, setCount] = useState<number>(0)
  const [searchcount, setSearchCount] = useState<number>(0)
  const [topag, setTopag] = useState<number>(0)
  const [topagfs, setTopagfs] = useState<number>(0)
  const [sort, setSort] = useState<string>("updated_at")
  const [order, setOrder] = useState<SortType>(SortType.D)
  const [isrsearch, setIsrsearch] = useState<boolean>(false)
  // const { loading: count_loading, data: count_data, error: count_error } = useQuery(GQL_GETCOUNT)
  const { data: count_data } = useQuery(GQL_GETCOUNT)
  const { data: list_data, refetch: refetchList } = useQuery(GQL_GETLIST, {
    fetchPolicy: 'network-only',
    variables: { page, take: 5, sort, order }
  })
  const { data: search_list_data } = useQuery(GQL_GETSEARCHLIST, {
    variables: {
      input: {
        page,
        take: 5,
        text: search
      }
    },
    skip: (search === "" || search.length <= 2)
  })

  // const [createTask, { loading, data, error }] = useMutation(GQL_CREATETASK, {
  //   onCompleted: refetchList
  // })
  const [createTask] = useMutation(GQL_CREATETASK, {
    onCompleted: refetchList
  })
  const [updateTask] = useMutation(GQL_UPDATETASK, {
    onCompleted: refetchList
  })
  const [deleteTask] = useMutation(GQL_DELETETASK, {
    onCompleted: refetchList
  })

  useEffect(() => {
    if (count_data) {
      setCount(count_data.count.count)
    }
  }, [count_data])

  useEffect(() => {
    if (list_data) {
      setTasks(list_data.list)
    }
  }, [list_data])

  useEffect(() => {
    if (search_list_data) {
      setSearchedTasks(search_list_data.searchList.data)
      setSearchCount(search_list_data.searchList.count)
    }
  }, [search_list_data])

  useEffect(() => {
    setTopag(calculatePagination(count))
  }, [count])

  useEffect(() => {
    setTopagfs(calculatePagination(searchcount))
  }, [searchcount])

  useEffect(() => {
    if (search === "") {
      setIsrsearch(false)
    }
  }, [search])

  const handleChange = (e: ChangeEvent) => {
    e.preventDefault()
    const day = new Date().getDay()
    const { value } = e.target as HTMLInputElement;
    setTask({
      title: days[day],
      text: value
    })
  }

  const handleSave = async (e: FormEvent) => {
    e.preventDefault()
    await createTask({ variables: { input: { title: task.title, text: task.text } } })
    setTask({
      title: "",
      text: ""
    })
  }

  const handleDelete = async (delid: number) => {
    await deleteTask({ variables: { id: delid } })
  }

  const handlePagination = async (cp: number) => {
    setPage(cp)
  }

  const handleSearch = async (e: FormEvent) => {
    e.preventDefault()
    setIsrsearch(true)
  }

  const handleOrder = async (val: string) => {
    switch (val) {
      case "ASC":
        setOrder(SortType.A)
        break;
      case "DESC":
        setOrder(SortType.D)
        break;
      default:
        break;
    }
  }

  const handleComplete = async (e: ChangeEvent, task: TaskType) => {
    const { checked } = e.target as HTMLInputElement;
    await updateTask({
      variables: {
        id: task.id,
        input: {
          title: task.title,
          text: task.text,
          completed: checked
        }
      }
    })
  }

  return (
    <div className="container mx-auto lg:my-32 md:my-30 sm:my-15 ">
      <InteractiveInputComponent handleSave={handleSave} handleChange={handleChange} task={task} search={search} setSearch={setSearch} setSort={setSort} handleSearch={handleSearch} handleOrder={handleOrder} />
      <DisplayTasks isrsearch={isrsearch} searchcount={searchcount} tasks={tasks} searchedTasks={searchedTasks} handleComplete={handleComplete} handleDelete={handleDelete} />
      <Pagination isrsearch={isrsearch} search={search} searchcount={searchcount} page={page} topag={topag} topagfs={topagfs} handlePagination={handlePagination} />
    </div>
  );
}

export default App;