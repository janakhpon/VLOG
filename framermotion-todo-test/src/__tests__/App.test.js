import { fireEvent, render, screen, within } from "@testing-library/react";
import App from "../App";

test("textarea should be in DOM", () => {
    render(<App />);
    const textboxElement = screen.getByRole("textbox");
    expect(textboxElement).toBeInTheDocument();
});

test("Task list should be in DOM", () => {
    render(<App />);
    const taskListElement = screen.getByRole("list");
    expect(taskListElement).toBeInTheDocument();
});

test("Task list should have 4 list items", () => {
    render(<App />);
    const taskListElement = screen.getByRole("list");
    const { getAllByRole } = within(taskListElement);
    const items = getAllByRole("listitem");
    expect(items.length).toBe(4);
});


test("Task list length should be increased after textarea form submit", () => {
    render(<App />);
    const textboxElement = screen.getByRole("textbox");
    fireEvent.focus(textboxElement);
    fireEvent.change(textboxElement, {
        target: { value: "Hello World" },
    });
    fireEvent.keyDown(textboxElement, {
        key: "Enter",
        code: "Enter",
        keyCode: 13,
        charCode: 13
    });
    const taskListElement = screen.getByRole("list");
    const { getAllByRole } = within(taskListElement);
    const items = getAllByRole("listitem");
    expect(items.length).toBe(5);
    const addedItemHeading = screen.getByRole('heading', { name: /hello world/i });
    expect(addedItemHeading).toBeInTheDocument();
});


test("Task list length should be decreased after clicked delete button", () => {
    render(<App />);
    const buttonElement = screen.getByTestId('itemButton0');
    fireEvent.click(buttonElement);
    const taskListElement = screen.getByRole("list");
    const { getAllByRole } = within(taskListElement);
    const items = getAllByRole("listitem");
    expect(items.length).toBe(4);
});

test("Container with 4 items snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
});