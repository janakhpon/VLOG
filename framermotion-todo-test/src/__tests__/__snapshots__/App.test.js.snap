// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container with 4 items snapshot 1`] = `
<div>
  <div
    class="h-screen bg-slate-300"
  >
    <div
      class="px-2 py-2 md:px-10 md:py-10 md:container md:mx-auto"
    >
      <div
        class="max-w-3xl mx-auto my-0 md:my-6"
      >
        <form
          class="w-full"
          data-projection-id="26"
          style="opacity: 0; transform: translateY(-100) translateZ(0);"
        >
          <textarea
            class=" form-control block w-full px-2 py-1.5 text-base font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
            id="exampleFormControlTextarea1"
            placeholder="Your new task"
            rows="4"
          />
        </form>
      </div>
      <ul
        class="overflow-y-auto"
        data-projection-id="27"
        data-testid="taskList"
        style="max-height: 80vh;"
      >
        <li
          class="max-w-3xl p-4 mx-auto my-4 rounded-lg shadow-lg bg-slate-200"
          data-projection-id="28"
          data-testid="listItem0"
          draggable="false"
          style="z-index: unset; opacity: 0; transform: translateX(-100px) translateY(0px) translateZ(0); user-select: none;"
        >
          <h4
            class="text-lg font-semibold text-gray-800"
          >
            Your mouth is a revolver, firing bullets in the sky (Mh-mh-mh-mh)
          </h4>
          <p
            class="mt-2 text-sm font-semibold text-gray-600"
          >
            Created on 
            Sat Jan 14 2022
          </p>
          <div
            class="flex justify-end space-x-2"
          >
            <svg
              class="w-6 h-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <svg
              class="w-6 h-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <svg
              class="w-6 h-6 cursor-pointer"
              data-testid="itemButton0"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </li>
        <li
          class="max-w-3xl p-4 mx-auto my-4 rounded-lg shadow-lg bg-slate-200"
          data-projection-id="29"
          data-testid="listItem1"
          draggable="false"
          style="z-index: unset; opacity: 0; transform: translateX(-100px) translateY(0px) translateZ(0); user-select: none;"
        >
          <h4
            class="text-lg font-semibold text-gray-800"
          >
            People like us, we dont Need that much, just some One that starts, starts the spark in our bonfire hearts
          </h4>
          <p
            class="mt-2 text-sm font-semibold text-gray-600"
          >
            Created on 
            Sat Jan 13 2022
          </p>
          <div
            class="flex justify-end space-x-2"
          >
            <svg
              class="w-6 h-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <svg
              class="w-6 h-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <svg
              class="w-6 h-6 cursor-pointer"
              data-testid="itemButton1"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </li>
        <li
          class="max-w-3xl p-4 mx-auto my-4 rounded-lg shadow-lg bg-slate-200"
          data-projection-id="30"
          data-testid="listItem2"
          draggable="false"
          style="z-index: unset; opacity: 0; transform: translateX(-100px) translateY(0px) translateZ(0); user-select: none;"
        >
          <h4
            class="text-lg font-semibold text-gray-800"
          >
            This world is getting colder, strangers passing by (Mh-mh-mh-mh) No one offers you a shoulder, no one looks you in the eye, eye, eye (Mh-mh-mh-mh)
          </h4>
          <p
            class="mt-2 text-sm font-semibold text-gray-600"
          >
            Created on 
            Sat Jan 15 2022
          </p>
          <div
            class="flex justify-end space-x-2"
          >
            <svg
              class="w-6 h-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <svg
              class="w-6 h-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <svg
              class="w-6 h-6 cursor-pointer"
              data-testid="itemButton2"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </li>
        <li
          class="max-w-3xl p-4 mx-auto my-4 rounded-lg shadow-lg bg-slate-200"
          data-projection-id="31"
          data-testid="listItem3"
          draggable="false"
          style="z-index: unset; opacity: 0; transform: translateX(-100px) translateY(0px) translateZ(0); user-select: none;"
        >
          <h4
            class="text-lg font-semibold text-gray-800"
          >
            Days like these lead to Nights like this lead to Love like ours You light the spark in my bonfire heart
          </h4>
          <p
            class="mt-2 text-sm font-semibold text-gray-600"
          >
            Created on 
            Sat Jan 12 2022
          </p>
          <div
            class="flex justify-end space-x-2"
          >
            <svg
              class="w-6 h-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <svg
              class="w-6 h-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <svg
              class="w-6 h-6 cursor-pointer"
              data-testid="itemButton3"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
